bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1128 @@
+    private final List<ChannelUID> legacyLinkedChannelUIDs = new CopyOnWriteArrayList<>();
+
+    private final ThingRegistry thingRegistry;
+    private final ItemChannelLinkRegistry itemChannelLinkRegistry;
+    private final Clip2StateDescriptionProvider stateDescriptionProvider;
+    private final ExecutorService threadPool = ThreadPoolManager.getPool("hue-api2-thing");
+
+    private String resourceId = "?";
+    private Resource thisResource;
+    private Duration dynamicsDuration = Duration.ZERO;
+    private Instant dynamicsExpireTime = Instant.MIN;
+
+    private boolean disposing;
+    private boolean hasConnectivityIssue;
+    private boolean updateSceneContributorsDone;
+    private boolean updateLightPropertiesDone;
+    private boolean updatePropertiesDone;
+    private boolean updateDependenciesDone;
+
+    private @Nullable ScheduledFuture<?> updateServiceContributorsTask;
+
+    public Clip2ThingHandler(Thing thing, Clip2StateDescriptionProvider stateDescriptionProvider,
+            ThingRegistry thingRegistry, ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing);
+
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (THING_TYPE_DEVICE.equals(thingTypeUID)) {
+            thisResource = new Resource(ResourceType.DEVICE);
+        } else if (THING_TYPE_ROOM.equals(thingTypeUID)) {
+            thisResource = new Resource(ResourceType.ROOM);
+        } else if (THING_TYPE_ZONE.equals(thingTypeUID)) {
+            thisResource = new Resource(ResourceType.ZONE);
+        } else {
+            throw new IllegalArgumentException("Wrong thing type " + thingTypeUID.getAsString());
+        }
+
+        this.thingRegistry = thingRegistry;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    /**
+     * Add a channel ID to the supportedChannelIdSet set. If the channel supports dynamics (timed transitions) then add
+     * the respective channel as well.
+     *
+     * @param channelId the channel ID to add.
+     */
+    private void addSupportedChannel(String channelId) {
+        if (!disposing && !updateDependenciesDone) {
+            synchronized (supportedChannelIdSet) {
+                logger.debug("{} -> addSupportedChannel() '{}' added to supported channel set", resourceId, channelId);
+                supportedChannelIdSet.add(channelId);
+                if (DYNAMIC_CHANNELS.contains(channelId)) {
+                    clearDynamicsChannel();
+                }
+            }
+        }
+    }
+
+    /**
+     * Clear the dynamics channel parameters.
+     */
+    private void clearDynamicsChannel() {
+        dynamicsExpireTime = Instant.MIN;
+        dynamicsDuration = Duration.ZERO;
+        updateState(CHANNEL_2_DYNAMICS, new QuantityType<>(0, MetricPrefix.MILLI(Units.SECOND)), true);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("{} -> dispose()", resourceId);
```

Comment:
It is useful for debugging. But I will shift it to 'trace'.

Reply:
Why is it useful when already logged by the framework?
```
2023-06-25 13:54:43.131 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:device:001788fffea2ef59:365d0a40-f11b-4aba-ad9f-0231921fe54d' changed from ONLINE to UNINITIALIZED
2023-06-25 13:54:43.135 [DEBUG] [e.internal.handler.Clip2ThingHandler] - 365d0a40-f11b-4aba-ad9f-0231921fe54d -> dispose()
```

And likewise for `initialize()`:
```
2023-06-25 13:54:45.122 [DEBUG] [e.internal.handler.Clip2ThingHandler] - 365d0a40-f11b-4aba-ad9f-0231921fe54d -> initialize()
2023-06-25 13:54:45.123 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:device:001788fffea2ef59:365d0a40-f11b-4aba-ad9f-0231921fe54d' changed from INITIALIZING to UNKNOWN
```

Reply:
Because the log messages `logger.trace("{} -> xyz", resourceId)` are prefixed with the resourceId specifically so one filter messages that relate to just one specific resource out of many -- see below..

```
2023-06-25 16:29:01.972 [DEBUG] [e.internal.handler.Clip2ThingHandler] - 1166743f-fe3d-47d1-bd7d-b5bb378098cc -> updateChannels() from resource id:468670da-d925-4b5b-bdd3-3c1387a9fd8f, type:grouped_light
2023-06-25 16:29:01.972 [DEBUG] [e.internal.handler.Clip2ThingHandler] - 1166743f-fe3d-47d1-bd7d-b5bb378098cc -> updateState() 'brightness' update with '0' (fullUpdate:false, isDefined:true)
2023-06-25 16:29:01.974 [DEBUG] [e.internal.handler.Clip2ThingHandler] - 1166743f-fe3d-47d1-bd7d-b5bb378098cc -> onResource() consumed resource id:468670da-d925-4b5b-bdd3-3c1387a9fd8f, type:grouped_light
```

Reply:
There are also occurrences like
```
    private synchronized void checkConnection() {
        logger.debug("checkConnection()");
```

If hunting such low level issues where a trace on method level is necessary, you might be more efficient when using the debugger.

The code base of this binding is big enough, we should not clutter it with debug code.

Our coding guidelines:
```
    Logging is not a replacement for using the debugger.

void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

Reply:
I hear what you say guys. But I am testing this binding not on a point basis in a debugger, but running it 7/24 on my live operative system. I can assure you that we would not have progressed as far we have, without massive and very detailed logging. This is a highly complex machine, so as the developer I do beg some indulgence.

Reply:
You made your points and I made my points. I don't think it's reasonable to stress this topic any further. I'm a fan of removing complexity, but if you feel more comfortable with leaving the debug messages, I'm okay with that, now.